src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -633,7 +637,7 @@ public bool DoesFragmentConditionMatch(ExecutionContext context, IHaveFragmentTy
                 return false;
             }
 
-            if (conditionalType.Equals(type))
+            if (conditionalType.ToString() == type.ToString())
```

Comment:
I see this was fixed since I started using the library source, it used to be `if (conditionalType == type)`. I'll revert to the code in master.